mbf_costmap_nav/include/mbf_costmap_nav/free_pose_search.h

```diff
@@ -197,6 +197,21 @@ class FreePoseSearch
    * @return A search solution for the given pose: best pose, state and cost
    */
   virtual SearchSolution search() const;
+
+private:
+  /**
+   * @brief Check if pose is collision-free; criteria: cost is not LETHAL, INSCRIBED and NO_INFORMATION
+   * @param cost
+   * @return true if cost is not LETHAL, INSCRIBED and NO_INFORMATION
+   */
+  bool isPoseValid(const unsigned char cost) const;
+
+  /**
+   * @brief Check if state is collision-free; criteria: state free or inscribed
```

Comment:
S/ F: Maybe rename `collision-free` to `non-lethal`. Coming from costmap terminology usually you'd consider inscribed cost as collision, but in this case we're not checking the center so it doesn't apply.